### ISSUE TITLE
No connection possible - E/A-Fehler: Connection to https:// ... refused

### ISSUE THREAD

**Issue by sh3wjie** [bitfireAT/davdroid#536]
Hi,
when trying to establish a connection to https://posteo.de:8443 as described on the DAVdroid as well as the posteo website I get a "E/A-Fehler: Connection to https://posteo.de:8443 refused" message. Playing around with some adaptions like adding a slash to the adress e.g. https://posteo.de:8443/ or adding @posteo.de to the username as proposed by posteo support did not help. Searching the I-Net also did not give a solution.
Although posteo support reports setting up DAVdroid on CM12.1 did work like a charm on one of their devices.
Maybe you have some hints for me how to analyse this problem.

Base data:
DAVdroid 0.8.0 from F-Droid
Motorola Moto G 2nd Gen. 2014 running CM 12.1-20150601-NIGHTLY-titan (Android 5.1.1)

Thank you for your efforts.

**Comment by untitaker** [comment]
Can you access that address in your browser?

**Comment by rfc2822** [comment]
I'm quite sure that this is not a DAVdroid problem. "Connection refused" means exactly that, so no TCP/IP connection is possible. Maybe a firewall or routing problem?

**Comment by sh3wjie** [comment]
Hi,
thx for this quick reply and the question/proposal to try the addres in a regular browser. Doing so leads to this message

Sabre_DAV_Exception_NotImplemented
  GET is only implemented on File objects
  1.7.1

For my understanding the "Connection refused" is a message the comes from an entity that prohibits to establish a network connection from my device. So I guess I have to dig deeper into this whole  topic.
Thank you so far. I will let you know when I have any further insights.

**Comment by rfc2822** [comment]
Do you use a privacy guard and have maybe revoked/not granted Internet access for DAVdroid?

**Comment by sh3wjie** [comment]
Not that I am aware of, but I must admit, I am a complete rookie when it comes to this whole smartphone/Android/mobile Apps thing because this is my first such device. What I know for sure is that I have a setting for data privacy for newly installed Apps set to active by default, so that I have to go there and deactivate it every time I do install a new App and want to grant access to my personal data. For DAVdroid I did deactivate it, so that DAVdroid has access. 
I will check that there are no other settings/Apps active that might interfere with DAVdroids attempts to connect to external servers. 
Thx for pointing me in that direction. I'll keep you in the loop.

**Comment by untitaker** [comment]
CM's privacy guard doesn't have any option to disable network access for apps, so this part should've worked. @sh3wjie did you try the browser on the same device, through the same ISP?

**Comment by sh3wjie** [comment]
**@rfc2822** : what I found out is that this privacy guard I was referring to seems to be a default in CM which basically is a shell to some underlying functionality.

**@untitaker** : good point! Yes the CM privacy guard only deals with other data privacy matters and not with any network access issues on the level of actually accessing a network as far as I can see.

Trying to connect to the server using one browser on the same device leads to a "Fehler: Verbindung fehlgeschlagen" message. Using another browser leads to"Webseite nicht verfügbar Die Webseite unter https://posteo.de:8443 konnte nicht geladen werden als: net::ERR_CONNECTION_REFUSED", which basically reads for me quite similar as to what DAVDroid tells me. In both cases the ISP was the same.
Question now is what keeps the browser(s)/DAVdroid App from connecting to the net? So far searching the I-NetI did not give an answer.

**Comment by untitaker** [comment]
Do you have any proxy set up? (check the WiFi settings in Android)
Are you able to access https://posteo.de:443?

**Comment by sh3wjie** [comment]
Well, checking the proxy settup under Einstellungen -> Drahtlos & Netzwerk -> WLAN -> Netzwerk ändern -> Erweitert reveals that there is no proxy set up (Keiner).
Checking the above mentioned address without the "8" leads to a redirection to https://posteo.de/de.

**Comment by rfc2822** [comment]
How do you access the Internet? WiFi only? Can you try mobile network, does it make a difference?

**Comment by sh3wjie** [comment]
Up to now I only did use WiFi. Changing to mobile network leads to the follwoing situation: both browsers now ask me to punch in my crendtials and after doing so I get a XML file presented that basically contains the error message of my 2nd post.
DAVdroid instead is now desperately trying to load data from the server. At least that is what it is telling me. "Daten werden vom Server abgefragt".

**Comment by sh3wjie** [comment]
Update 1: after changing to mobile networks, closing all Apps and browsers and starting anew to set up an DAVdroid account I was able to get past the screen where you type in your access data in DAVdroid an now I have a DAvdroid account! There are 2 things I have to figure out now:
1. How to use DAVdroid in detail
2. Why doesn't it work when I am trying to access the I-Net with WiFi?

Update 2:
Just read a few line on the DAVdroid website and did as was described there. I opened the regular calendar and there I saw all my events and appointments that I did enter in the posteo calendar.

PERFECT!!!! A dream!

So now I have the proof that it basically works with mobile access. This leaves me with question 2. of Update 1. So somehow something is bend with my WiFi settings. I will look deeper into that.

Thank you.

**Comment by rfc2822** [comment]
Ok, so I'll close this now as it's not a DAVdroid problem. If you find a solution for your problem, you can post it here anyway of course.

**Comment by sh3wjie** [comment]
It's ok with me. Thank you again for your patience and support. I'll let
you know the outcome of my further investigation.

--
Mit freundlichem Gruß,
Hertwig

Am 07.06.2015 um 15:16 schrieb rfc2822:

**Comment by sh3wjie** [comment]
DER GEHT!!!! 
It works!!!! After another hint from the posteo support I checked my router settings. There I have set up two accesses, a main access and one guest access. So far I was using the guest access to access the net. Checking the settings there showed a possibility to restrict I-Net access only to surfing and mailing. After unchecking the respective check box I tried to use the previously set up DAVdroid account to synch my data and ... IT WORKS!!!

So now I am a happy DAVdroid user and posteo customer. What else can I ask for?

Thank you so much for your time and patience in helping me analysing this problem although it was on my side and not on yours. Sorry for any inconvenience.

Just wanted to let you know.

Thanks again. Topic can stay closed.
